packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -49,20 +49,20 @@ export type CheckboxSlots = {
    * The root slot receives the `className` and `style` specified directly on the `<Checkbox>`.
    * All other native props will be applied to the primary slot: `input`
    */
-  root: NonNullable<Slot<'span'>>;
+  root: Slot<'span'>;
 
   /**
    * The Checkbox's label.
    */
-  label?: Slot<typeof Label>;
+  label?: Slot<typeof Label> | null;
 
   /**
    * Hidden input that handles the checkbox's functionality.
    *
    * This is the PRIMARY slot: all native properties specified directly on `<Checkbox>` will be applied to this slot,
    * except `className` and `style`, which remain on the root slot.
    */
-  input: NonNullable<Slot<'input'>>;
+  input: Slot<'input'>;
 
   /**
    * The checkbox, with the checkmark icon as its child when checked.
```

Comment:
Note: the `indicator` slot was previously nullable, but with this change it is non-nullable (as a result of `Slot` no longer including `| null`). It appears from the code that it was never intended to be nullable.